train.c

```diff
@@ -0,0 +1,94 @@
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
+#include <float.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include "agents/temporal_difference.h"
+#include "game.h"
+#include "util.h"
+
+
+static double epsilon = 0.3;
+#define NUM_ITERATION 10000
+#define CALC_STATE_NUM(x)                 \
+    {                                     \
+        x = 1;                            \
+        for (int i = 0; i < N_GRIDS; i++) \
+            x *= 3;                       \
+    }
```

Comment:
This macro is defined once in temporal_difference.h and once in train.c. Is it possible to consolidate them so that it is only defined once?